geode-core/src/main/java/org/apache/geode/management/internal/cli/functions/CacheRealizationFunction.java

```diff
@@ -77,14 +79,23 @@ public void execute(FunctionContext<List> context) {
     }
   }
 
-  public Object executeGet(InternalCache cache,
+  public RuntimeInfo executeGet(FunctionContext<List> context,
+      InternalCache cache,
       CacheElement cacheElement) {
     ConfigurationRealizer realizer = realizers.get(cacheElement.getClass());
 
     if (realizer == null) {
       return null;
     }
-    return realizer.get(cacheElement, cache);
+    RuntimeInfo runtimeInfo = realizer.get(cacheElement, cache);
+
+    // set the membername if this is not a global runtime
+    if (cacheElement instanceof CorrespondWith
```

Comment:
Would it be possible to have CacheElement implement CorrespondWith so we wouldn't need to do this "instanceof"?

Reply:
not until we've implemented v2 API support for *all* CacheElements.  we did previously try to eliminate this instanceof check in #3714 but decided against it

Reply:
After we separate the config objects from the xml domain objects, we probably can combine CacheElement, RestfulEndPoint and CorrespondWith all together, even though they each serve a different purpose, but we don't have a case where one config object would implement one without others.